src/engine/sparqlExpressions/PrefilterExpressionIndex.cpp

```diff
@@ -539,38 +540,46 @@ BlockMetadataRanges PrefixRegexExpression::evaluateImpl(
   static_assert(Vocab::PrefixRanges::Ranges{}.size() == 1);
   using LVE = LocalVocabEntry;
   LocalVocab localVocab{};
-  std::string_view prefixSv(&prefix_.front(), prefix_.size());
-  const auto& prefixRanges = vocab.prefixRanges(prefixSv);
-  const auto& ranges = prefixRanges.ranges();
-  const auto& [lowerVocabIndex, upperVocabIndex] = ranges.front();
+  auto prefixQuoted =
+      absl::StrCat("\"", asStringViewUnsafe(prefixLiteral_.getContent()));
+  auto [lowerVocabIndex, upperVocabIndex] =
+      vocab.prefixRanges(prefixQuoted).ranges().front();
 
   // Set lower reference.
   const auto& lowerIdVocab = Id::makeFromVocabIndex(lowerVocabIndex);
-  const auto& lowerIdLocalVocab = getValueIdFromIdOrLocalVocabEntry(
-      LVE::fromStringRepresentation(absl::StrCat(prefix_, "\"")), localVocab);
-  auto lowerId =
-      lowerIdVocab < lowerIdLocalVocab ? lowerIdVocab : lowerIdLocalVocab;
+  const auto& beginIdIri = getValueIdFromIdOrLocalVocabEntry(
+      LVE::fromStringRepresentation("<"), localVocab);
 
   if (isNegated_) {
+    const auto& upperIdAdjusted =
```

Comment:
We need some comments here, maybe I will write them myself, if that is the only issue that remains.